packages/core/src/routes/interaction/consent.ts

```diff
@@ -201,12 +200,84 @@ export default function consentRoutes<T extends IRouterParamContext>(
 
       const userInfo = await queries.users.findUserById(accountId);
 
-      const { missingOIDCScope, missingResourceScopes } = getMissingScopes(prompt);
+      const { missingOIDCScope, missingResourceScopes: allMissingResourceScopes = {} } =
+        getMissingScopes(prompt);
+
+      // The missingResourceScopes from the prompt details are from `getResourceServerInfo`,
+      // which contains resource scopes and organization resource scopes.
+      // We need to separate the organization resource scopes from the resource scopes.
+      // The "scopes" in `missingResourceScopes` do not have "id", so we have to rebuild the scopes list.
+      const missingResourceScopes = await parseMissingResourceScopesInfo(
```

Comment:
Should we update the `parseMissingResourceScopesInfo` method?  Put all the logic inside. 

```ts
const missingResourceScopes = await parseMissingResourceScopesInfo(queries, allMissingResourceScopes);

const organizationsWithMissingResourceScopes = await Promise.all(
  organizations.map(async ({ name, id }) => {
     const missingResourceScopes = parseMissingResourceScopesInfo(queries, allMissingResourceScopes, id);
     return { name, id, missingResourceScopes };
}));
```

Reply:
Updated